updates for tts dac model

mmwillet opened the pull request:
Adds simple support for TTS audio decoding.

Commit:
updates for tts dac model

Commit:
cruft

Commit:
additionally changes for original TTS model

Commit:
adds support for ggml_reciprocal

Commit:
adds metal shader support for reciprocal and conv transpose functions

Commit:
support conv_transposition with 16bit weights

Commit:
more reasonable gguf tensor name buffer size

Commit:
adds support for common ops used by kokoro

Commit:
removes complex requirement

Commit:
cleans up thread assignment logic and fixes overflow in istft

Commit:
fixes reflective end padding in STFT and improper linear interpolation implementation

Commit:
fixes istft overflow

danielzgtg commented:
I would like to rebase in the latest changes from upstream. I will squash just the "cruft" commits.

By doing so, the ggml_op enum numbers at https://github.com/mmwillet/ggml/pull/1/files#diff-a2e9aefaa59b1831c998cfb7d6adf0d6a4974912c985445e9fca48f9f4602facR548-R556 will need to be increased. I hope that will not break compatibility with the existing gguf files uploaded to Hugging Face. If it does, then we should upstream earlier.

mmwillet commented:
>By doing so, the ggml_op enum numbers at https://github.com/mmwillet/ggml/pull/1/files#diff-a2e9aefaa59b1831c998cfb7d6adf0d6a4974912c985445e9fca48f9f4602facR548-R556 will need to be increased. I hope that will not break compatibility with the existing gguf files uploaded to Hugging Face. If it does, then we should upstream earlier.

It will not. Those enums are purely for internal GGML use (as far as I know) and incrementing them presents no current compatibility issues assuming GGML is not being run as a sidecar (or some other async process protocol).

danielzgtg committed:
Enable vectorization for ggml_compute_forward_sin

Kokoro perf_battery before=1804.631546ms after=1685.014389ms (-6.63%)

With -fopenmp and -ffast-math, GCC replaces sinf@PLT with the
vectorized version _ZGVdN8v_sinf@PLT .

Commit:
Merge pull request #2 from mmwillet/feat/sin-ffast-math

Enable vectorization for ggml_compute_forward_sin

mmwillet committed:
adds depthwise conv1d

danielzgtg committed:
cherry-pick: C++17

From d50637e1b50ac71c3cabb4752408f02b8d66c203

Needed for my mmap work
